### ISSUE TITLE
Make starting up a second isolate group deprecated

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#115507]
Previously, it was required in some use cases to create multiple flutter engines that didn't share resources because dart isolate groups didn't exist and platform channels were unable to be used from background isolates.  Now that both are possible via FlutterEngineGroup APIs and Isolate Platform Channels; we should deprecate the use of generating a second isolate group.  Doing so can result in: unexpected behaviors from Dart (ex https://github.com/flutter/flutter/issues/113825), worse performance and there is no known reason why it should be done.

That means that anytime someone creates a new dart isolate group when there is one already in existence it should result in a loud warning in the terminal (since there is no way to block it at build time).  We should also add a deprecation warning on FlutterEngine's constructor, encouraging usage of FlutterEngineGroup instead.  Lastly, Android and iOS apps should use FlutterEngineGroups by default.

In summary:
1) [ ] Print out error message when creating second live dart isolate group
1) [ ] Deprecate FlutterEngine constructor on iOS and Android
1) [ ] Start using FlutterEngineGroup in iOS and Android by default

**Comment by endless7** [comment]
Hi - @gaaclarke, I'm interested in this issue. Can I get started on it?

**Comment by gaaclarke** [comment]
hi @endless7, let's' wait a bit.  This is more of a proposal and I want to get more feedback from the team that we want to do this.

**Comment by gaaclarke** [comment]
Looks like `android_alarm_manager_plus` also spawns a background engine in a different isolate group: https://github.com/gaaclarke/plus_plugins/blob/7e898675fbd18cb6ac90f36c1e6bd126aea1c64d/packages/android_alarm_manager_plus/android/src/main/java/dev/fluttercommunity/plus/androidalarmmanager/FlutterBackgroundExecutor.java#L158:L158

**Comment by mkustermann** [comment]
>  Make FlutterEngineGroup singletons

There may be add2app scenarios where users want to use different flutter apps (i.e. different AOT-compiled apps) that render different screens. To share common code, such users should ideally compile one bit AOT app with all sub-apps inside and use different `main` entrypoints for the sub-apps. Though it's unclear if users always do that (and/or we make it easy enough to do that).
=> One may decide to have a `Map<AOT-asset, FlutterEngineGroup>`

The problem I discovered is that it's really difficult for current creators of flutter engine (groups) to get a reference to an existing group: The reason is that flutter plugins may register with OS to be called for background execution when certain events happen. When that plugin's native code is then invoked it will want to execute dart code, but there's no way for it to know whether there's an existing engine (apart from making `FlutterEngine` / `FlutterEngineGroup` have a global registry of active engines). I wonder how such plugins even know which AOT-asset to use when creating an flutter engine (group), anyone knows?
